packages/openactive-integration-tests/test/features/approval/minimal-proposal/implemented/accept-proposal-book-test.js

```diff
@@ -31,101 +30,68 @@ FeatureHelper.describeFeature(module, {
   // even if some OrderItems don't require approval, the whole Order should
   controlOpportunityCriteria: 'TestOpportunityBookable',
 },
-(configuration, orderItemCriteria, featureIsImplemented, logger, state, flow) => {
-  beforeAll(async () => {
-    await state.fetchOpportunities(orderItemCriteria);
+(configuration, orderItemCriteria, featureIsImplemented, logger) => {
+  // ## Initiate Flow Stages
+  const fetchOpportunities = FetchOpportunitiesFlowStage.create({ orderItemCriteria, logger });
```

Comment:
I did play about with this @nickevansuk. Great minds! It requires a little bit of figuring out for this test as one of the stage descriptions use facts of previous stages i.e. the action uses a thunk which uses `p.getResponse()`.

But it would be easy to make a function which does that stitching together for simpler tests which do not have requirements like the above.

And it would also be possible to make a more flexible one which supports the above case. I'll try to include something at a later stage of this PR

Reply:
Another related thought, what if we used the chaining pattern here to set `prerequisite` and `initiatedBy` automatically? e.g.

```javascript
const flowSet = FlowFactory.createFlow(logger)
  .addStage('FetchOpportunities', FetchOpportunitiesFlowStage, { orderItemCriteria })
  .addStage('C1', C1FlowStage)
  .addStage('C2', C2FlowStage)
  .addStage('TestInterface`, TestInterfaceActionFlowStage, {
     testName: 'Simulate Seller Approval (Test Interface Action)',
     createActionFn: () => ({
       type: 'test:SellerAcceptOrderProposalSimulateAction',
       objectType: 'OrderProposal',
       objectId: p.getResult().body['@id'],
     })
  .addStage('OrderFeedUpdate', OrderFeedUpdateFlowStage, C2FlowStage)
  .addStage('B', BFlowStage)
```

That would make it slightly magic, however would mean we only need to reference the logger once, and different stages could then be accessed via the key in the resulting `Set`. We could even default the Key, so `  .addStage('C1', C1FlowStage)` could become `.addStage(C1FlowStage)`?

Might be more concise given it'll be repeated so many times? Might also provide more flexibility than the array pattern as params can be added to the stages.

Reply:
@nickevansuk I'm going to resolve this for now as I'm going to do the top-level magic function in a separate PR